src/QueryDescription/test.js

```diff
@@ -479,6 +506,87 @@ describe('QueryDescription', () => {
           },
         },
       ],
+      sortBy: [],
+      skip: null,
+      take: null,
+    })
+  })
+  it('supports sorting query', () => {
+    const query = Q.buildQueryDescription([
+      Q.sortBy('sortable_column', 'desc'),
+    ])
+    expect(query).toEqual({
+      where: [],
+      join: [],
+      sortBy: [
+        {
+          type: 'sortBy',
+          sortColumn: 'sortable_column',
+          sortOrder: 'desc',
+        },
+      ],
+      skip: null,
+      take: null,
+    })
+  })
+  it('supports take operator', () => {
+    const query = Q.buildQueryDescription([
+      Q.take(100),
+    ])
+    expect(query).toEqual({
+      where: [],
+      join: [],
+      sortBy: [],
+      take: {
+        type: 'take',
+        count: 100,
+      },
+      skip: null,
+    })
+  })
+  it('does not support skip operator without take operator', () => {
+    expect(() => {
+      Q.buildQueryDescription([
+        Q.skip(100),
+      ])
+    }).toThrow('cannot skip without take')
+  })
+  it('supports multiple take operators and take the last', () => {
+    const query = Q.buildQueryDescription([
+      Q.take(100),
+      Q.take(200),
+      Q.take(400),
+    ])
+    expect(query).toEqual({
+      where: [],
+      join: [],
+      sortBy: [],
+      take: {
+        type: 'take',
+        count: 400,
+      },
+      skip: null,
+    })
+  })
+  it('support multiple skip operators but only take the last', () => {
+    const query = Q.buildQueryDescription([
+      Q.take(100),
+      Q.skip(200),
+      Q.skip(400),
+      Q.skip(800),
+    ])
+    expect(query).toEqual({
+      where: [],
+      join: [],
+      sortBy: [],
+      take: {
+        type: 'take',
+        count: 100,
+      },
+      skip: {
+        type: 'skip',
+        count: 800,
+      },
```

Comment:
and also locate the test for Query.extend and add a test for that

Reply:
see https://github.com/Nozbe/WatermelonDB/pull/622/files#diff-cc4cc709a149f36c03a3875eb540a4afR84